src/template/wasm2c.declarations.c

```diff
@@ -89,20 +89,22 @@ static inline bool func_types_eq(const wasm_rt_func_type_t a,
   (CHECK_CALL_INDIRECT(table, ft, x),       \
    DO_CALL_INDIRECT(table, t, x, __VA_ARGS__))
 
-#ifdef SUPPORT_MEMORY64
+#if __has_builtin(__builtin_add_overflow)
+#define add_overflow(a, b, resptr) __builtin_add_overflow(a, b, resptr)
+#else /* MSVC */
```

Comment:
Seems a little fragile to assume all compilers without `__builtin_add_overflow` must be MSVC.

How about `#else if defined(MSVC) ... #else error`

Reply:
I wonder if we should also consider a simple C-only fallback, although it maybe fine to separate this change.

Reply:
Good idea... done.